src/font.rs

```diff
@@ -0,0 +1,173 @@
+use crate::stream::Stream;
+use crate::Error;
+use std::fmt::Display;
+
+/// Parses a list of font families and generic families from a string.
+pub fn parse_font_families(text: &str) -> Result<Vec<FontFamily>, Error> {
+    let mut s = Stream::from(text);
+    let font_families = s.parse_font_families()?;
+
+    s.skip_spaces();
+    if !s.at_end() {
+        return Err(Error::UnexpectedData(s.calc_char_pos()));
+    }
+
+    Ok(font_families)
+}
+
+/// A type of font family.
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub enum FontFamily {
+    /// A serif font.
+    Serif,
+    /// A sans-serif font.
+    SansSerif,
+    /// A cursive font.
+    Cursive,
+    /// A fantasy font.
+    Fantasy,
+    /// A monospace font.
+    Monospace,
+    /// A custom named font.
+    Named(String),
+}
+
+impl Display for FontFamily {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
+            FontFamily::Monospace => "monospace".to_string(),
+            FontFamily::Serif => "serif".to_string(),
+            FontFamily::SansSerif => "sans-serif".to_string(),
+            FontFamily::Cursive => "cursive".to_string(),
+            FontFamily::Fantasy => "fantasy".to_string(),
+            FontFamily::Named(s) => format!("\"{}\"", s),
+        };
+        write!(f, "{}", str)
+    }
+}
+
+impl<'a> Stream<'a> {
+    pub fn parse_font_families(&mut self) -> Result<Vec<FontFamily>, Error> {
+        let mut families = vec![];
+
+        while !self.at_end() {
+            self.skip_spaces();
+
+            let family = {
+                let ch = self.curr_byte()?;
+                if ch == b'\'' || ch == b'\"' {
+                    let res = self.parse_quoted_string()?;
+                    FontFamily::Named(res.to_string())
+                } else {
+                    let mut idents = vec![];
+
+                    while !self.at_end() {
+                        let c = self.chars().next().unwrap();
```

Comment:
Can crash on multi-byte character.

Reply:
Do you have an example? 😅 Since we are ensuring that we are not at the end there always is at least one character remaining, no?

Reply:
`at_end` checks that we have at least one **byte**, but if the byte is not ASCII, `chars` will return `None`. I guess it's not possible to trigger in this case since we guarantee a valid string, but I would still suggest using:

```rust
while let Some(c) = self.chars().next() {
```

Reply:
> I guess it's not possible to trigger in this case since we guarantee a valid string, but I would still suggest using

Right, I was assuming that the string is valid. But Rust guarantees that it's valid UTF-8, and the current position should always be at the start of a character, no? That was exactly why I consciously chose unwrap here instead of this, because in theory it should never happen. So if for some reason it does happen, isn't it better to let it crash (so that it can be reported more easily) instead of trying to handle it gracefully, which means that it might never be discovered?

Reply:
Ideally, there should be 0 unwrap in the resvg and all of its dependencies. Everything can happen, that's the problem.

Reply:
Right, I'll fix it then.